lib/reply.js

```diff
@@ -553,6 +553,7 @@ function handleError (reply, error, cb) {
       })
     : serializerFn(Object.create(error, {
       error: { value: statusCodes[statusCode + ''] },
+      code: { value: error.code },
       message: { value: error.message || '' },
       statusCode: { value: statusCode }
```

Comment:
Your test passes to the `handleError` function an `error` that looks like:

```js
{
  message: 'kaboom',
  statusCode: 400,
  code: 'FST_ERR_HELLO_WORLD'
}
```

Thus, when this condition is hit, the `Object.create(error, { /* additional properties */ })` will create a new object to pass into the serializer who's prototype includes a `code` property. The schema in the test looks for the `code` property, finds it on the prototype, and fills it in correctly.

The following test passes on the `master` branch (i.e. without this PR applied):

```js
"use strict";

const fastify = require("fastify")();

fastify.get(
  "/",
  {
    schema: {
      response: {
        500: {
          type: "object",
          properties: {
            code: { type: "string" },
            error: { type: "string" },
            message: { type: "string" },
            statusCode: { type: "integer" },
          },
          required: ["code", "error", "message", "statusCode"],
          additionalProperties: false,
        },
      },
    },
  },
  (req, res) => {
    const error = new Error("boom");
    error.statusCode = 500;
    error.code = "FST_ERR_HELLO_WORLD";
    res.serializer((data) => JSON.stringify(data)).send(error);
  }
);

fastify.inject({ url: "/", method: "GET" }, (err, res) => {
  if (err) console.error(err);
  console.log(res.json());
});
```

What conditions necessitate this PR? Without a replication in the test, this PR does not seem needed.

Reply:
It looks like the error is something specifically to do with `fastify-error`. When I do

```js
import createError from 'fastify-error';
const CustomError = createError('FST_ERR_HELLO_WORLD', 'error message', 500);

// ...

reply.send(new CustomError());
```
I can consistently reproduce. Is this still in scope for `fastify/fastify` or should an issue be raised on `fastify/fastify-error`?